### DISCUSSION TITLE
FieldSlicer error

### DISCUSSION THREAD

**Discussion by mncrowe** [CliMA/Oceananigans.jl#3210] [Computational science!] [answered]
Hi,

I've been trying to set up an output writer to save surface fields for a 3D simulation.

From what I've read, the following should work:

simulation.output_writers[:surface_slice_writer] = NetCDFOutputWriter(model, merge(model.tracers, (; KE, p, zeta));
                                                      schedule = TimeInterval(1hours),
                                                      filename = "eady_turbulence_512_surf",
                                                      overwrite_existing = true,
                                                      field_slicer = FieldSlicer(k=grid.Nz))

However, I get the error: "UndefVarError: `FieldSlicer` not defined". Also, printing the output writer gives:

NetCDFOutputWriter scheduled on TimeInterval(12 hours):
├── filepath: ./eady_turbulence_512.nc
├── dimensions: zC(64), zF(65), xC(128), yF(128), xF(128), yC(128), time(0)
├── 5 outputs: (v, w, b, u, zeta)
└── array type: Array{Float64}

which suggests it doesn't have field_slicer as a parameter. I've attached the full code below.

Does anyone know what I'm doing wrong here?

Thanks,

Matt

[eady.jl.txt](https://github.com/CliMA/Oceananigans.jl/files/12251872/eady.jl.txt)

**Comment by navidcy** [comment, marked as answer]
Please include which version of Oceananigans you are using. I have a feeling that you may be using deprecated syntax ie syntax that does not correspond to the version of Oceananigans you have installed. `indices` has replaced FieldSlicer quite a while now I think. See, eg,

https://clima.github.io/OceananigansDocumentation/stable/generated/baroclinic_adjustment/#Diagnostics/Output

and

https://clima.github.io/OceananigansDocumentation/stable/appendix/library/#Oceananigans.OutputWriters.NetCDFOutputWriter-Tuple{Any,%20Any}

**Reply by mncrowe** [reply]
Thanks, version is v0.86.0.Uusing 'indices' fixes this as you say. 

Matt
